### DISCUSSION TITLE
use_rows in pf.add_parameters

### DISCUSSION THREAD

**Discussion by wkitlasten** [pypest/pyemu#412] [General]
Can use_rows be a string to identify the row based on the index column, or was I dreaming (#sad). Otherwise, we have to parse the input file separately just to get the row number?

**Comment by jtwhite79** [comment]
I think you are dreaming...haha.  @briochh would know for sure...

**Comment by briochh** [comment]
@wkitlasten, you may not be in a complete dream state. You should be able to pass detail on the rows to set up based on the value of index col -- if you pass use_rows as > 1D object (list of whatever). e.g. `use_rows=[(1)]` should set up only where the index col is `1`. Note, (I believe) that this is the value of the index_col in the model input file (i.e. before we do any zero-based adjustments in pyemu)... I think it should also work for string values ie `use_rows=[("this_one"), ("and_this_one")]`. Interested to know how that goes for you.

**Comment by wkitlasten** [comment]
haha, I can't even get: use_rows= [('par1'), ('par2')]

use_rows = ['par1', 'par2']

On Thu, Mar 2, 2023 at 4:04 PM Brioch Hemmings ***@***.***>
wrote:

> @wkitlasten <https://github.com/wkitlasten>, you may not be in a complete
> dream state. You should be able to pass detail on the rows to set up based
> on the value of index col -- if you pass use_rows as > 1D object (list of
> whatever). e.g. use_rows=[(1)] should set up only where the index col is 1.
> Note, (I believe) that this is the value of the index_col in the model
> input file (i.e. before we do any zero-based adjustments in pyemu)... I
> think it should also work for string values ie use_rows=[("this_one"),
> ("and_this_one")]. Interested to know how that goes for you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pypest/pyemu/discussions/412#discussioncomment-5174136>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADSJXRGSLSDWSHIMDOFC4G3W2AE4HANCNFSM6AAAAAAVMZDRXI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
"Perfect spheres are pointless."
